### PR TITLE
Rhmap 9456 missing credential field

### PR DESCRIPTION
**Pull request by alanmoran**
When using fhc to build apps (`fhc build`) it results in the "credential" field in studio build history being left empty. If you complete the same build via the studio UI the field is populated. This is due to the credentialName param not being populated in FHC.

Had to create a lookup to find it. Thought about allowing the user to specific the param itself but I think it would be confusing to allow the user to call the credential whatever they wanted.

### CODE REVIEW

File: lib/common.js [hunk 1]
```diff
@@ -984,6 +984,19 @@ function endsWith(str, suffix) {
   return str.indexOf(suffix, str.length - suffix.length) !== -1;
 }
 
+function findCredentialsByID(id, cb) {
+	exports.doGetApiCall(fhreq.getFeedHenryUrl(), "box/api/credentials", i18n._("Error reading credentials: "), function(err, data) {
```
Comment (fh-qube) [inline lib/common.js]: MINOR Replace all tab characters in this file by sequences of white-spaces. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATabCharacter)

File: lib/common.js [hunk 2]
```diff
@@ -984,6 +984,19 @@ function endsWith(str, suffix) {
...
+		if (err) {
+			return cb(err);
+		}
+		var result = _.find(data, function(o) {
```
Comment (fh-qube) [inline lib/common.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

File: lib/cmd/common/build.js [hunk 3]
```diff
@@ -198,42 +198,55 @@ function mutateGitRefArgs(args) {
...
-      }
-      var keys = [];
-      if (data.cacheKey) keys.push(data.cacheKey);
-      //      if(data.stageKey) keys.push(data.stageKey);
-      if (keys.length > 0) {
-        async.map(keys, common.waitFor, function(err, results) {
+
+    //Fix missing creds names - RHMAP-9456
+    if (args.bundleId) {
+      common.findCredentialsByID(args.bundleId, function(err, data) {
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

File: lib/cmd/common/build.js [hunk 4]
```diff
@@ -198,42 +198,55 @@ function mutateGitRefArgs(args) {
 function doBuild(args, cb) {
   var uri = "box/srv/1.1/wid/" + fhc.curTarget + "/" + args.destination + "/" + args.app + "/deliver";
   var doCall = function() {
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MINOR Function has a complexity of 22 which is greater than 20 authorized. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: lib/cmd/common/build.js [hunk 5]
```diff
@@ -198,42 +198,55 @@ function mutateGitRefArgs(args) {
...
-                results.push({
-                  download: data
+              if (results[0] && results[0][0] && results[0][0].action) {
+                var build_asset = results[0][0].action.url;
+                build.message = i18n._("\nDownload URL: ") + build_asset;
+                downloadBuild(args, build_asset, './', function(err, data) {
+                  if (err) {
+                    return cb(err);
+                  }
+                  if (data) {
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

File: lib/cmd/common/build.js [hunk 6]
```diff
@@ -198,42 +198,55 @@ function mutateGitRefArgs(args) {
...
                 return cb(err);
               }
-              if (data) {
-                results.push({
-                  download: data
+              if (results[0] && results[0][0] && results[0][0].action) {
+                var build_asset = results[0][0].action.url;
+                build.message = i18n._("\nDownload URL: ") + build_asset;
+                downloadBuild(args, build_asset, './', function(err, data) {
+                  if (err) {
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

**Comment by fh-qube** [comment]
SonarQube analysis reported 6 issues:
* BLOCKER 2 blocker
* MAJOR 2 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* The code changes conform to the contributing & code style guides for this component (if they exist)
* A component owner should review your change (we've tagged these above)
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.
Fhcap PR will be generated automatically.
To deploy your changes to a cluster place comment in this PR with pattern [x][y] where x=deploy and y=clustername you wish to deploy. 
You can also deploy your change manually using fhcap-cli

**Comment by fh-qube** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: lib/cmd/common/build.js [hunk 7]
```diff
@@ -48,7 +48,9 @@ function build(argv, cb) {
       validateArgsNgui(argObj);
     }
     mutateGitRefArgs(argObj);
-    doBuild(argObj, cb);
+    addCredentialsParams(argObj, function(){
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.

File: lib/cmd/common/build.js [hunk 8]
```diff
@@ -195,6 +197,23 @@ function mutateGitRefArgs(args) {
...
+  //Fix missing creds names - RHMAP-9456
+  if (args.bundleId) {
+    common.findCredentialsByID(args.bundleId, function(err, data) {
+      if (!err) {
+        //If this is not set its just set as blank
+        if (data.bundleName) {
+          args.credentialName = data.bundleName;
+        }
+      }
+      return cb();
```
Comment (wei-lee) [inline lib/cmd/common/build.js]: we should probably pass err back if there is any

Comment (alanmoran) [inline lib/cmd/common/build.js]: @wei-lee I was thinking the same and was torn by this one. It's not a mandatory field so the build works with or without the credential name being passed. I wasn't sure if I should be blocking the build if this "label" call fails. More than happy to update if you think I should.

**Comment by alanmoran** [comment]
PR rejected as per internal notes. This will be fixed inside core to add lookup

**Pull request closed without merging**